sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -56,6 +56,20 @@ class JDBCSuite extends QueryTest
       Some(StringType)
   }
 
+  val testH2DialectTinyInt = new JdbcDialect {
+    override def canHandle(url: String): Boolean = url.startsWith("jdbc:h2")
+    override def getCatalystType(
+        sqlType: Int,
+        typeName: String,
+        size: Int,
+        md: MetadataBuilder): Option[DataType] = {
+      sqlType match {
+        case java.sql.Types.TINYINT => Some(ByteType)
```

Comment:
I'm wondering if this is okay in this `H2` dialect context.
We made `testH2DialectTinyInt` like this, but theoretically, `TINYINT` is still unsigned and Spark `ByteType` is signed.

Reply:
Yeah, I think we found similar issue in another PR. @twdsilva can you check and fix it if needed?